recon-enumeration/index.md

```diff
@@ -139,7 +139,65 @@ The total number of requests generated in the attack is the product of the numbe
 There are also many different types of payloads you can use ([here](https://portswigger.net/burp/documentation/desktop/tools/intruder/payloads/types) is the entire list), from specifying your own list of words to generating random bytes.
 You can find lists of popular credentials online, for instance, [here](https://github.com/danielmiessler/SecLists/tree/master/Passwords) is a repo with lists of most used passwords.
 
-## Web Content Enumeration/ Web Content Discovery/ Dirbusting
+## Testing for default credentials
+
+Using the system's default credentials is a relatively easy and popular technique to gain first access.
+The default non-random passwords that come with many devices (especially those in the Internet of Things) are frequently left untouched.
+
+You can read more about this [here](https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/04-Authentication_Testing/02-Testing_for_Default_Credentials).
+You can also find wordlists that have all the default credentials for most known vendors in one place, such as [this one](https://github.com/ihebski/DefaultCreds-cheat-sheet).
+
+## Types of brute force attacks
+
+There are many different types of performing a brute force attack, but we will mention just three.
+
+### Dictionary attack
+
+This is a straighforward technique and maybe the default one that comes to mind when trying to discover a set of (username, password) credentials.
+Given a wordlist of usernames and a wordlist of passwords, for all the usernames, all the passwords are tried.
+
+Example:
+```
+usernames = [user1, user2]
+passwords = [pass1, pass2]
+We try, in this order: (user1, pass1), (user1, pass2), (user2, pass1), (user2, pass2)
+```
+
+### Password-spray attack
+
+This technique is a bit more complicated, but it is suitable for situations where we want to avoid username lockout.
+That is, when the target blocks us after a number of wrong attempts for a username and, for a certain period of time, we can't make any more attempts.
+
+Taking this into account, the process is as follows:
+- We take one password at a time and try all usernames for it (instead of taking a username and trying all passwords, like for Dictionary attack).
+- At the beginning we define a constant representing the _maximum number of attempts to try the same username_ in a period.
+- After this limit is reached, we introduce a _delay_ (which can be viewed as the lockout period) and perform a "sleep".
+- After this, we resume the attack.
+
+Example:
+```
+usernames = [user1, user2]
+passwords = [pass1, pass2]
+username_attempts_per_period = 1
+lockout_period = 5
+We try, in this order: (pass1, user1), (pass2, user2), sleep(5), (pass1, user2), (pass2, user2)
```

Comment:
This example is a bit off, as stated above, we should try the first password for every username, sleep, then try the next password for every single user and so on ...
The example should be:
"We try, in this order:  (pass1, user1), (pass1, user2), sleep(5), (pass2, user1), (pass2, user2)"